xbmc/windows/GUIMediaWindow.cpp

```diff
@@ -282,8 +282,8 @@ bool CGUIMediaWindow::OnMessage(CGUIMessage& message)
       else if (iControl == CONTROL_BTNSORTBY) // sort by
       {
         if (m_guiState.get())
-          m_guiState->SetNextSortMethod();
-        UpdateFileList();
+        if (m_guiState->ChooseSortMethod())
```

Comment:
merge to one `if`. is nesting them like this even correct? I can never remember:\